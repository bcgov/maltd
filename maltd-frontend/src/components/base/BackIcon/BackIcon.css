#main {
  position: absolute;
  padding: 10px;
  cursor: pointer;
}

#main > div {
  float: right;
}

.size-large {
  font-size: 40px;
}

.limit-width {
  width: 100px;
}
