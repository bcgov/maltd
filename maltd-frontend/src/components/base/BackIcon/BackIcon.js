import React from "react";
import PropTypes from "prop-types";
import { FaAngleLeft } from "react-icons/fa";
import "./BackIcon.css";

export default function BackIcon({ backIcon: { message }, onClick }) {
  return (
    <div
      id="main"
      role="button"
      tabIndex="0"
      onClick={onClick}
      onKeyDown={onClick}
    >
      <FaAngleLeft className="size-large" />
      <div className="limit-width">
        <h5>Back</h5>
        <p>{message}</p>
      </div>
    </div>
  );
}

BackIcon.propTypes = {
  backIcon: PropTypes.shape({
    message: PropTypes.string.isRequired
  }).isRequired,
  onClick: PropTypes.func.isRequired
};
