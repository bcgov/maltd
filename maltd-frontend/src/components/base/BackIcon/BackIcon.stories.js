/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import BackIcon from "./BackIcon";

const backIcon = {
  message: "Find another user"
};

storiesOf("BackIcon", module).add("default", () => (
  <BackIcon backIcon={backIcon} onClick={action("back button clicked")} />
));
