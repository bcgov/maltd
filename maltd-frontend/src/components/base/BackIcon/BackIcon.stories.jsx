import React from "react";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import BackIcon from "./BackIcon.jsx";

const backIcon = {
  message: "Find another user"
};

storiesOf("BackIcon", module).add("default", () => (
  <BackIcon backIcon={backIcon} onClick={action("back button clicked")} />
));
