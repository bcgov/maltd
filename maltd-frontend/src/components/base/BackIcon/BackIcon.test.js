import React from "react";
import renderer from "react-test-renderer";
import BackIcon from "./BackIcon";

describe("Back Icon", () => {
  test("Component renders as expected", () => {
    const component = renderer.create(
      <BackIcon
        backIcon={{
          message: "Find another user"
        }}
        onClick={() => jest.fn()}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
