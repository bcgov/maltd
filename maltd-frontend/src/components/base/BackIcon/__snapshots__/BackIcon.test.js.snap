// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Icon Component renders as expected 1`] = `
<div
  id="main"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <svg
    className="size-large"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 256 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
    />
  </svg>
  <div
    className="limit-width"
    data-cy="back-nav"
  >
    <p>
      Find another user
    </p>
  </div>
</div>
`;
